jenkins/bootstrap.py

```diff
@@ -303,6 +306,11 @@ def finish(gsutil, paths, success, artifacts, build, version, repo):
         except subprocess.CalledProcessError:
             logging.warning('Failed to write build path to %s', paths.pr_path)
 
+    meta = metadata(repo, artifacts)
+    if not version:
+        version = meta.get('job-version')
```

Comment:
The way metadata is currently generated, I think only underscores are supported. (We might be able to change this, though.)

I'm also not sure about the name `job-version`, though I don't immediately have a better suggestion.
For reference, the upgrade and skew tests set `kubernetes_skew_version`, which also isn't great.